provider/spark.go

```diff
@@ -1401,11 +1401,56 @@ func (d *DatabricksExecutor) SparkSubmitArgs(destPath filestore.Filepath, cleanQ
 	credentialConfigs := store.CredentialsConfig()
 	argList = append(argList, credentialConfigs...)
 
-	argList = append(argList, "--source_list")
-	argList = append(argList, sourceList...)
+	// Databricks's API enforces a 10K-byte limit on job submit params, so to avoid a 400, we need to check to ensure
+	// the args are below this limit. If they exceed this limit, it's most likely due to the query and/or the list of
+	// sources, so we write these as a JSON file and read them from the PySpark runner script to side-step this constraint
+	if d.exceedsSubmitParamsTotalByteLimit(argList, cleanQuery, sourceList) {
```

Comment:
nit: should we just throw this and under it into a sep function?

`handleSubmitParams` or something